gslib/tests/testcase/base.py

```diff
@@ -102,8 +102,8 @@ def MakeTempName(self, kind, prefix=''):
       providers (e.g. replacing "_" with "-", converting uppercase letters to
       lowercase, etc.).
     """
-    name = '{prefix}gsutil-test-{method}-{kind}'.format(
-      prefix=prefix, method=self.GetTestMethodName(), kind=kind)
+    name = '{prefix}gsutil-test-{method}-{kind}{suffix}'.format(
+      prefix=prefix, method=self.GetTestMethodName(), kind=kind, suffix=suffix)
```

Comment:
With all these additions, it's possible to go over the `MAX_BUCKET_LENGTH-9` character cutoff that we do below; it's possible that this completely chops off the suffix, breaking assumptions that the test might have based on the suffix of the bucket name.

We might want to have a sanity check here that fails if suffix was passed in and part of it was cut off, e.g.:

```
    name = '{prefix}gsutil-test-{method}-{kind}{suffix}'.format(
        prefix=prefix, method=self.GetTestMethodName(), kind=kind, suffix=suffix)
    if suffix and kind == 'bucket':
      # Chop off 9 characters so we can append a random string and
      # psuedo-randomize the bucket name and not go over the length limit.
      old_name = name
      name = name[:MAX_BUCKET_LENGTH-9]
      # Fail if we would have chopped off all or part of the suffix, as
      # test logic might depend on the suffix being present.
      if len(name) > len(old_name):
        self.fail(
            'Tried to create a psuedo-random bucket name with a specific '
            'suffix, but the generated name was too long and there was not '
            'enough room for the suffix. Please use shorter strings or perform '
            'name randomization manually.\nRequested name: %s\nName truncated '
            'to make room for random string: %s' % (old_name, name))

    name = '{name}-{rand}'.format(name=name, rand=self.MakeRandomTestString())
```

Reply:
Got it, that makes sense to me! Would it make sense if I put this logic in the `MakeBucketNameValid` method, since that gets called for `'bucket' type objects in this logic?

Likewise, would it be reasonable to also send the prefix and suffix to `MakeBucketNameValid` and then if the name is too long, automatically truncate from the base bucket name enough characters until the name is small enough, and only throw an error if `len(prefix) + len(suffix) >= MAX_BUCKET_LENGTH-12` or something? Or if that risks loosing too much entropy and causing bucket name collisions, I'll just do the logic you have above. :)

Reply:
Actually, I realized `MakeBucketName` isn't from the testing side of the library, I'll just do this here in base.py. 🙂

Reply:
Actually I just realized that the suffix is used exactly zero times when calling this method. Maybe it would be simplest if I just remove the suffix.